### PR TITLE
chore(Milestone Details): Upgrade from AngularJS to Angular

### LINKED ISSUE #955: chore(Milestone Details): Upgrade from AngularJS to Angular
### Description

Upgrade MilestoneDetails component from AngularJS to Angular and clean up code.

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Notes
- @breity please edit styling as you see fit, in particular:
   - Milestone report graphs appear vertically instead of horizontally inline
   -  "Student completion" view alignment is off
Screenshot 2022-12-19 at 9 45 42 AM

## Changes
- Register MilestoneReportGraph/Data as WebComponents to get them to render dynamically in authored content
- Moved all milestone* component declarations to ClassroomMonitorModule for now
   - Will make a separate PR to break up ClassroomMonitorModule into separate modules. Doing this in the PR would have created too many file changes and expanded scope
- Clean up code

## Test Prep
- Run ```npm install``` to get dependency

## Test
- Milestone report works as before
   - report graph and data are computed and rendered properly 
- If report is...
   - available: can view/grade student work
   - not available: can view student progress
- Step info displays in another popup dialog

Closes #955

### CODE REVIEW

**Review by breity — CHANGES_REQUESTED** [review]
I updated and fixed the styling issues.

This looks good for the most part, but viewing student work when there are multiple locations for a milestone is broken now.

Screen Shot 2023-01-12 at 3 48 03 PM

I see some js errors related to the milestone graphs, but the graphs do seem to render properly. I also see a server error:
```
wise-api     | java.lang.NullPointerException
wise-api     | 	at rufus.lzstring4java.LZString.getBaseValue(LZString.java:36)
wise-api     | 	at rufus.lzstring4java.LZString.access$300(LZString.java:16)
wise-api     | 	at rufus.lzstring4java.LZString$6.doFunc(LZString.java:123)
wise-api     | 	at rufus.lzstring4java.LZString._decompress(LZString.java:404)
wise-api     | 	at rufus.lzstring4java.LZString.decompressFromEncodedURIComponent(LZString.java:120)
wise-api     | 	at org.wise.vle.web.wise5.TeacherGetDataController.getComponentsList(TeacherGetDataController.java:82)
wise-api     | 	at org.wise.vle.web.wise5.TeacherGetDataController.getData(TeacherGetDataController.java:61)
...
```

**Comment by hirokiterashima** [comment]
@breity looks like we just needed to merge the latest from develop into this branch (because your API was using a new way to parse components param). PTAL.

File: src/assets/wise5/classroomMonitor/classroomMonitorComponents/milestones/milestone-details/milestone-details.component.html [hunk 1]
```diff
@@ -0,0 +1,96 @@
+
```
Comment (geoffreykwan) [inline src/assets/wise5/classroomMonitor/classroomMonitorComponents/milestones/milestone-details/milestone-details.component.html]: This file has an empty new line at the top which can be deleted.

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
Switching between periods in the Milestone dialog won't update the graph data.

**Comment by breity** [comment]
@hirokiterashima Multiple locations works now that the latest from develop has been merged. ✅ 

I see the same issue with switching periods that @geoffreykwan identified (MilestoneReportGraph and MilestoneReportData instances don't update).

**Comment by hirokiterashima** [comment]
Fixed the issue updating graph/data when period changes. PTAL.

**Review by geoffreykwan — APPROVED** [review]
Looks good.

**Review by breity — APPROVED** [review]
LGTM.

**Pull request merged**